doc/api/deprecations.md

```diff
@@ -2247,6 +2247,21 @@ Type: Documentation-only (supports [`--pending-deprecation`][])
 Directly calling `process.binding('uv').errname(<val>)` is deprecated.
 Please make sure to use [`util.getSystemErrorName()`][] instead.
 
+<a id="DEPXXXX"></a>
+### DEPXXXX: tls Server.prototype.setOptions()
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/XXXXX
+    description: Runtime deprecation.
+-->
+
+Type: Runtime
+
+The `tls` module's `Server.prototype.setOptions()` method was undocumented, and
+not meant to be used outside of Node.js core. Use
+`Server.prototype.setSecureContext()` instead.
```

Comment:
Optional nit: Don't bother with the explanation, as it's a bit of a non-explanation anyway? Just tell people what they should use?

```markdown
Please use `Server.prototype.setSecureContext()`.
```